sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -30,9 +30,13 @@ def test_piecewise():
         Piecewise((x, Or(x < 1, x < 2)), (0, True))
     assert Piecewise((x, x < 1), (x, x < 2), (x, True)) == x
     assert Piecewise((x, True)) == x
+    # Empty Piecewise not accepted
+    raises(TypeError, lambda: Piecewise())
     # False condition is never retained
-    assert Piecewise((x, False)) == Piecewise(
-        (x, False), evaluate=False) == Piecewise()
+    assert Piecewise((2*x, x < 0), (x, False)) == \
+        Piecewise((2*x, x < 0), (x, False), evaluate=False) == \
+        Piecewise((2*x, x < 0))
+    raises (TypeError, lambda: Piecewise((x, False)))
```

Comment:
I'm not sure about this one. I think it would be better for this case to also return nan, and only do TypeError for explicit `Piecewise()` with no arguments. It's possible for piecewise like these to be constructed, for instance, if a condition evaluates to False. Plus the error message would be confusing in this case.